utils/node_local_test.py

```diff
@@ -3231,6 +3233,144 @@ def test_daos_fs_check(self):
         if dir_nr != 12:
             raise NLTestFail('Wrong number of sub-directories in lost+found')
 
+    def test_daos_fs_fix(self):
+        """Test DAOS FS Fix Tool"""
+
+        pool = self.pool.id()
+        cont = self.container
+
+        dfuse = DFuse(self.server,
+                      self.conf,
+                      pool=self.pool.id(),
+                      container=self.container,
+                      caching=False)
+        dfuse.start(v_hint='fs_fix_test', log_mask='CRIT')
+        path = dfuse.dir
+        dirname = join(path, 'test_dir')
+        os.mkdir(dirname)
+
+        fname1 = join(dirname, 'f1')
+        with open(fname1, 'w') as fd:
+            fd.write('test1')
+        fname2 = join(dirname, 'f2')
+        with open(fname2, 'w') as fd:
+            fd.write('test2')
+
+        dirname1 = join(path, 'test_dir/1d1/')
+        os.mkdir(dirname1)
+        fname3 = join(dirname1, 'f3')
+        with open(fname3, 'w') as fd:
+            fd.write('test3')
+        dirname2 = join(path, 'test_dir/1d2/')
+        os.mkdir(dirname2)
+        fname4 = join(dirname2, 'f4')
+        with open(fname4, 'w') as fd:
+            fd.write('test4')
+
+        # start corrupting things
+        daos_mw_fi = join(self.conf['PREFIX'], 'lib/daos/TESTING/tests/', 'daos_mw_fi')
+        cmd_env = get_base_env()
+        cmd_env['DAOS_AGENT_DRPC_DIR'] = self.conf.agent_dir
+        cmd = [daos_mw_fi, pool, cont, "corrupt_entry", "/test_dir/f1"]
+        print(cmd)
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=dfuse.dir)
+        assert rc.returncode == 0
+        cmd = [daos_mw_fi, pool, cont, "corrupt_entry", "/test_dir/1d1/f3"]
+        print(cmd)
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=dfuse.dir)
+        assert rc.returncode == 0
+        cmd = [daos_mw_fi, pool, cont, "corrupt_entry", "/test_dir/1d2"]
+        print(cmd)
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=dfuse.dir)
+        assert rc.returncode == 0
+
+        # try to read from corrupted entries. all should fail
+        try:
+            fd = os.open(fname1, 'r')
+            os.close(fd)
+            assert False
+        except TypeError:
+            pass
```

Comment:
`open` is expected to fail, right? And `close` is just in case it doesn't? You might consider this approach, which will handling closing for you. 
```suggestion
        try:
            with os.open(fname1, 'r'):
                assert False
        except TypeError:
            pass
```

Reply:
If we don't need a reference to the handle, then we don't need `as fd`

Reply:
> What I'm wondering about here though is why TypeError? That seems like a python error and I'd expect a OSError of some kind.

im not sure.. don't have time on my side to dig into python specifics, but it does report the appropriate error from the C API side.

Reply:
Ah, this isn't right. Consider this:
```
>>> import os
>>> fd = open('foo', 'r')
>>> fd
<_io.TextIOWrapper name='foo' mode='r' encoding='UTF-8'>
>>> os.close(fd)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: an integer is required (got type _io.TextIOWrapper)
```
What you want is `fd.close()`, which `with open...` does for you. But if you're getting a `TypeError` is probably means the `open` is succeeding here.

Reply:
To use `os.close`, you would need to also use `os.open`

Reply:
python --

Reply:
but thanks @daltonbohning I think i have it right this time :-)